src/App.tsx

```diff
@@ -1,39 +1,102 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
 
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
+
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
```

Comment:
clear listeners in componentWillUnmount
![image](https://user-images.githubusercontent.com/70009205/233132623-5a8486d2-9564-479f-9786-e0d0582892d3.png)